protocol/message_handler.go

```diff
@@ -337,6 +337,17 @@ func (m *MessageHandler) HandleChatMessage(state *ReceivedMessageState) error {
 	state.ModifiedChats[chat.ID] = true
 	state.AllChats[chat.ID] = chat
 
+	contact := state.CurrentMessageState.Contact
+	if overrideENSName(contact, receivedMessage.EnsName, receivedMessage.Clock) {
```

Comment:
This would probably make more sense as `hasENSNameChanged()`, and maybe add comment like `This should trigger ENS re-verification`.